sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1996,7 +1996,7 @@ class Dataset[T] private[sql](
   def dropDuplicates(colNames: Seq[String]): Dataset[T] = withTypedPlan {
```

Comment:
you have not updates docs for dropDuplicates! You should at least point to withWatermark to limit the state, and mention its semantics (all data later than watermark will be ignored).